src/coreclr/nativeaot/Common/src/Internal/Runtime/MethodTable.cs

```diff
@@ -268,15 +269,38 @@ internal static bool SupportsWritableData
         private const uint ValueTypePaddingAlignmentMask = 0xF8;
         private const int ValueTypePaddingAlignmentShift = 3;
 
+        internal bool HasComponentSize
+        {
+            get
+            {
+                return ((_usFlags & (ushort)EETypeFlags.HasComponentSizeFlag) != 0);
+            }
+#if TYPE_LOADER_IMPLEMENTATION
+            set
+            {
+                if (value)
+                {
+                    _usFlags |= (ushort)EETypeFlags.HasComponentSizeFlag;
```

Comment:
Should we just set the flag as part of setting the ComponentSize?

Reply:
I felt like it would be less prone to mistakes if by the time when we set component size we would need to indicate whether we are going to have component size or flags.

That is just in case we set extended flags and then overwrite with a component size. 
It is hard to make such mistake if you need to set HasComponentSize accordingly before setting either extended flags or ComponentSize.

Reply:
ok